astropy/time/formats.py

```diff
@@ -611,6 +617,29 @@ def set_jds(self, val1, val2):
         jd1 = self.epoch.jd1 + day
         jd2 = self.epoch.jd2 + frac
 
+        # For the usual case that scale is the same as epoch_scale, we only need
```

Comment:
This feels a bit inelegant, and I fear may break round-to-even. But it brings up that the above addition is not really proper either as it can loose a last bit in the fraction unnecessarily and what we'd really want is to be able to add two (jd1,jd2) pairs safely, using that we know that those are already int and frac, so that for the `jd1` addition there is no worry about precision loss. This would help addition and subtraction too...

My overall sense would be, for this PR, to just use `day_frac` and then raise an issue for doing that sum (I may try to get something together now).

Reply:
Well, we're testing round to even in `test_precision` now, though possibly not for these time formats. 
But I must admit is partially is simply having so much extra code just for the speed-up. Is there anything against just doing
```
jd1_extra = np.round(jd2) #  Maybe add: if day.shape else round(jd2[()])
jd1 += jd1_extra
jd2 -= jd1_extra
```

Reply:
Ah, that's embarrassing I missed such an obvious and simple solution. BTW, it turns out that `jd2.item()` is about 3x faster than `jd2[()]`, so that is what I'm using.